packages/lesswrong/components/sunshineDashboard/SunshineNewTagsItem.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { Components, registerComponent } from '../../lib/vulcan-lib';
+import { useUpdate } from '../../lib/crud/withUpdate';
+import { Tags } from '../../lib/collections/tags/collection';
+import Users from '../../lib/collections/users/collection';
+import { Link } from '../../lib/reactRouterWrapper'
+import { useCurrentUser } from '../common/withUser';
+import { useHover } from '../common/withHover'
+import DoneIcon from '@material-ui/icons/Done';
+import ClearIcon from '@material-ui/icons/Clear';
+import withErrorBoundary from '../common/withErrorBoundary'
+import { commentBodyStyles, } from '../../themes/stylePiping'
+import { useMulti } from '../../lib/crud/withMulti';
+import { TagRels } from '../../lib/collections/tagRels/collection';
+
+const styles = theme => ({
+  tagInfo: {
+    ...commentBodyStyles(theme),
+    marginTop: 0,
+    marginBottom: 0
+  },
+  postCount: {
+    ...theme.typography.commentStyle,
+    ...theme.typography.smallText,
+    marginTop: 12,
+    marginBottom: 8,
+    color: theme.palette.grey[600]
+  },
+  post: {
+    ...theme.typography.body2,
+    ...theme.typography.postStyle,
+    marginBottom: 4,
+    color: theme.palette.grey[700]
+  }
+})
+
+const SunshineNewTagsItem = ({tag, classes}: {
+  tag: SunshineTagFragment,
+  classes: ClassesType
+}) => {
+  const currentUser = useCurrentUser();
+  const {eventHandlers, hover, anchorEl} = useHover();
+  
+  const {mutate: updateTag} = useUpdate({
+    collection: Tags,
+    fragmentName: 'SunshineTagFragment',
+  });
+
+
+
+  const handleApprove = () => {
+    updateTag({
+      selector: { _id: tag._id},
+      data: {
+        reviewedByUserId: currentUser!._id,
```

Comment:
Why are you using an exclamation point here? I think you should add a null-check to the beginning of this function if you want the type-system to do what you want.

Reply:
That looks like a straight-up error, probably not supposed to be there at all. But, adding a null check seems good.

Reply:
Oh, this was copy-pasted from other Sunshine items. I think it's plausible they should all change, but I think I'd prefer to do that in a separate pass? I don't actually know what the exclamation point does.

Reply:
(looks like Jim added them to other Sunshine elements, you may want to chat with Jim and get into agreement on preferred approach here)